src/hooks/useSourceExpansion.js

```diff
@@ -0,0 +1,50 @@
+import {useReducer, useEffect, useMemo} from 'react';
+import {expandFromRawCode} from '../utils';
+
+const useAppendableCollection = () => {
```

Comment:
Should we merge `useCollection` (in `useChangeSelect`) and `useAppendableCollection` to form a powerful helper hook?

Reply:
This reminds me that we need some description to hooks. At least.

Reply:
Yes, maybe also some implement from our own application?